test/functional/legacy/eval_spec.lua

```diff
@@ -495,6 +495,18 @@ describe('eval', function()
       ' abcE4b10-4\000abcE4b10-4-2')
   end)
 
+  it("setreg('0',getreg('a',1,1)) doesn't crash for unset register", function()
```

Comment:
No, wrong. Do not use `setreg` here *at all*, test that `getreg('a', 1, 1) isnot# v:_null_list`. Or that it can be extended.